src/main/scala/com/actian/spark_vector/datastream/DataStreamClient.scala

```diff
@@ -47,17 +51,36 @@ case class DataStreamClient(vectorProps: VectorConnectionProperties, table: Stri
   private def prepareUnloadSql(table: String) = s"prepare for x100 stream from $table"
   private def startUnloadSql(selectQuery: String) = s"insert into external table $selectQuery"
 
-  private def executeSql(sql: String): Future[Int] = Future { jdbc.executeStatement(sql) }
+  /** Execute a sql (statement) within a future task */
+  private def executeSql(sql: String): Future[Int] = Future { jdbc.executeStatement(sql) } match {
+    case f =>
```

Comment:
doing a match without extracting stuff is not really worth it. how about {
val f = Future { ... }
f onFailure ...
f
}